R/fitNullModel.R

```diff
@@ -176,3 +179,12 @@ nullModelInvNorm <- function(null.model, cov.mat = NULL,
         stop("all sample names in dimnames of cov.mat must be present in x$sample.id")
     }
 }
+
+
+## Return a small version of the null model (no NxN matrices)
+smallNullModel <- function(null.model) {
+    null.model$cholSigmaInv <- NULL
+    null.model$CX <- NULL
+    null.model$CXCXI <- NULL
+    null.model
+}
```

Comment:
Also buried in a comment responding to @mconomos's review:
- Should we add a list element indicating if the null model is small? then we can check for the existence of that list element instead of checking for all the required elements.

Alternatively, turn null models into S4 objects - a base class, and two classes that extend that base class (eg fullNullmodel and smallNullModel) - and use signatures. This is a much bigger chance.

Reply:
Adding the `isNullModelSmall` function should solve this problem in a simple way. It doesn't even need to check for all the elements, just `cholSigmaInv`.